### ISSUE TITLE
[Bug]: During multi-GPU training, all cards occupy 100% and freeze

### ISSUE THREAD

**Issue by zuowanbushiwo** [speechbrain/speechbrain#2055]
### Describe the bug
After training an epoch, it gets stuck in the middle of the second epoch, There is no problem with training with a single card (with command : `CUDA_VISIBLE_DEVICES=3  python train.py  conf_yaml_json/train_tspnn_subband.yaml`), Almost always, but sometimes in other epochs, but in the middle of the training process, not the beginning, not the end, and not the evaluate process。
I think the difference between my code and your templates is that  I define a compute_loss function in my model class, It is because my loss function needs to use some variables and functions in the model class.
```
def compute_objectives(self, predictions, batch, stage):
    if stage == sb.Stage.TRAIN :
        loss = self.modules.model.module.compute_loss(predictions["wav"], target_sig, predictions["other"])
    else:
        loss = self.hparams.compute_si_snr(predictions["wav"], target_sig, lens)
```

![image](https://github.com/speechbrain/speechbrain/assets/3354428/88984c16-9b9f-4ce7-bbbf-5031bc2cce52)

![image](https://github.com/speechbrain/speechbrain/assets/3354428/c53b69fa-b124-4b98-b298-23e4d1e8c0cc)

Trian commad：
`CUDA_VISIBLE_DEVICES=0,1,2,3   python -m torch.distributed.launch --nproc_per_node=4 train.py conf_yaml_json/train_tspnn_subband.yaml  --distributed_launch  --find_unused_parameters`

```
absl-py                   1.4.0
aiohttp                   3.8.4
aiosignal                 1.3.1
alembic                   1.9.4
antlr4-python3-runtime    4.8
appdirs                   1.4.4
asteroid                  0.6.0
asteroid-filterbanks      0.4.0
asttokens                 2.2.1
async-timeout             4.0.2
attrdict                  2.0.1
attrs                     22.2.0
audioread                 3.0.0
backcall                  0.2.0
backports.cached-property 1.0.2
brotlipy                  0.7.0
brouhaha                  0.9.0
cached-property           1.5.2
cachetools                5.3.0
certifi                   2022.12.7
cffi                      1.15.1
charset-normalizer        2.0.4
click                     8.1.3
cmaes                     0.9.1
colorama                  0.4.6
coloredlogs               15.0.1
colorlog                  6.7.0
comm                      0.1.2
commonmark                0.9.1
conda                     23.1.0
conda-package-handling    2.0.2
conda_package_streaming   0.7.0
contourpy                 1.0.7
cryptography              38.0.4
cycler                    0.11.0
Cython                    0.29.34
debugpy                   1.6.6
decorator                 5.1.1
DeepFilterDataLoader      0.4.0
DeepFilterLib             0.4.0
deepfilternet             0.4.0
docopt                    0.6.2
einops                    0.3.2
et-xmlfile                1.1.0
exceptiongroup            1.1.1
executing                 1.2.0
filelock                  3.9.0
flatbuffers               23.1.21
flit_core                 3.6.0
fonttools                 4.38.0
frozenlist                1.3.3
fsspec                    2023.1.0
future                    0.18.3
google-auth               2.16.1
google-auth-oauthlib      0.4.6
greenlet                  2.0.2
grpcio                    1.51.3
hmmlearn                  0.2.8
huggingface-hub           0.12.1
humanfriendly             10.0
Hydra                     2.5
hydra-core                1.1.0
HyperPyYAML               1.1.0
icecream                  2.1.3
idna                      3.4
importlib-metadata        6.0.0
importlib-resources       5.12.0
iniconfig                 2.0.0
ipyhton                   0.1
ipykernel                 6.21.2
ipython                   8.10.0
jedi                      0.18.2
joblib                    1.2.0
julius                    0.2.7
jupyter_client            8.0.3
jupyter_core              5.2.0
kiwisolver                1.4.4
librosa                   0.9.2
llvmlite                  0.39.1
loguru                    0.7.0
Mako                      1.2.4
Markdown                  3.4.1
MarkupSafe                2.1.2
matplotlib                3.7.0
matplotlib-inline         0.1.6
MinDAEC                   0.0.2
mir-eval                  0.7
mkl-fft                   1.3.1
mkl-random                1.2.2
mkl-service               2.4.0
mpmath                    1.2.1
multidict                 6.0.4
multiprocessing-logging   0.3.4
natsort                   8.3.1
nest-asyncio              1.5.6
networkx                  2.8.8
numba                     0.56.4
numpy                     1.23.5
oauthlib                  3.2.2
omegaconf                 2.1.2
onnx                      1.13.1
onnx-simplifier           0.4.17
onnxruntime               1.14.0
openpyxl                  3.1.2
openyxl                   0.1
optuna                    3.1.0
packaging                 23.0
pandas                    1.5.3
parso                     0.8.3
pb-bss-eval               0.0.2
pesq                      0.0.4
pexpect                   4.8.0
pickleshare               0.7.5
Pillow                    9.3.0
pip                       22.3.1
pipdeptree                2.5.0
platformdirs              3.0.0
pluggy                    1.0.0
pooch                     1.6.0
primePy                   1.3
prompt-toolkit            3.0.37
protobuf                  3.20.3
psutil                    5.9.4
ptyprocess                0.7.0
pure-eval                 0.2.2
pyannote.audio            2.1.1
pyannote.core             4.5
pyannote.database         4.1.3
pyannote.metrics          3.2.1
pyannote.pipeline         2.3
pyasn1                    0.4.8
pyasn1-modules            0.2.8
pybind11                  2.10.4
pycosat                   0.6.4
pycparser                 2.21
pyDeprecate               0.3.2
Pygments                  2.14.0
pyOpenSSL                 22.0.0
pyparsing                 3.0.9
pyreadline                2.1
pyroomacoustics           0.7.3
PySocks                   1.7.1
pystoi                    0.3.3
pytest                    7.3.1
python-dateutil           2.8.2
pytorch-lightning         1.6.5
pytorch-metric-learning   1.7.3
pytorch-ranger            0.1.1
pytz                      2022.7.1
pyvad                     0.2.0
PyYAML                    6.0
pyzmq                     25.0.0
requests                  2.28.1
requests-oauthlib         1.3.1
resampy                   0.4.2
rich                      12.6.0
rsa                       4.9
ruamel.yaml               0.17.21
ruamel.yaml.clib          0.2.7
scikit-learn              1.2.1
scipy                     1.10.1
semver                    2.13.0
sentencepiece             0.1.97
setuptools                65.6.3
shellingham               1.5.0.post1
simplejson                3.18.3
singledispatchmethod      1.0
six                       1.16.0
sortedcontainers          2.4.0
SoundFile                 0.10.3.post1
speechbrain               0.5.13
SQLAlchemy                2.0.4
stack-data                0.6.2
sympy                     1.11.1
tabulate                  0.9.0
tensorboard               2.12.0
tensorboard-data-server   0.7.0
tensorboard-plugin-wit    1.8.1
threadpoolctl             3.1.0
toml                      0.10.2
tomli                     2.0.1
toolz                     0.12.0
torch                     1.11.0
torch-audiomentations     0.11.0
torch-optimizer           0.1.0
torch-pitch-shift         1.2.2
torch-stoi                0.1.2
torch-tb-profiler         0.4.1
torchaudio                0.11.0
torchinfo                 1.7.2
torchmetrics              0.7.3
torchsummary              1.5.1
torchvision               0.12.0
tornado                   6.2
tqdm                      4.64.1
traitlets                 5.9.0
typer                     0.7.0
typing_extensions         4.4.0
urllib3                   1.26.14
wcwidth                   0.2.6
webrtcvad                 2.0.10
Werkzeug                  2.2.3
wheel                     0.38.4
yarl                      1.8.2
zipp                      3.14.0
zstandard                 0.19.0
```

### Expected behaviour

normal operation

**Comment by TParcollet** [comment]
Are all the gpu on a single machine? It looks like a DDP issue related to communication.

**Comment by wennyramadha** [comment]
Hi @zuowanbushiwo how did you solve the problem? I also got the same issue.
